src/Core/Utility.re

```diff
@@ -230,3 +230,27 @@ let ranges = indices =>
     indices,
   )
   |> List.rev;
+
+// TODO: Remove after 4.08 upgrade
+module Option = {
+  let map = f =>
+    fun
+    | Some(x) => Some(f(x))
+    | None => None;
+
+  let value = (~default) =>
+    fun
+    | Some(x) => x
+    | None => default;
+
+  let some = x => Some(x);
+
+  let bind = f =>
+    fun
+    | Some(x) =>
+      switch (f(x)) {
+      | None => None
+      | Some(_) as v => v
+      }
+    | None => None;
```

Comment:
Beware that the argument order here is the wrong way around compared to `Stdlib.Option.bind`. It's weird, but apparently that's the convention for `bind`. A `flat_map` function was initially also included with the arguments in the"right" order, but that was taken out because it traditionally originated on the `List` type or something, and it would apparently be weird to have it in `Option` and not in `List`.

Reply:
That's good to know! I didn't realize that. I would've expected this is the 'right way' because it is so amenable for piping.

I guess, once we upgrade to 4.08, and pick up the `Stdlib.Option.bind`, we'd switch these to use the monadic-let anyway?

Reply:
Yeah, that would be nice. I hope I can get used to them quickly though, as they're not exactly intuitive. Maybe Reason will provide a better syntax for it.